src/test/java/org/llorllale/cactoos/matchers/TextMatcherEnvelopeTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import java.io.StringReader;
+import org.cactoos.Text;
+import org.cactoos.text.TextOf;
+import org.junit.Test;
+
+/**
+ * Tests for {@link TextMatcherEnvelope}.
+ * @since 1.0.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle JavadocTypeCheck (500 lines)
+ */
+public final class TextMatcherEnvelopeTest {
+
+    @Test
+    public void matchesAReadOnceInput() {
+        final String input = "aaaa";
+        new Assertion<>(
+            "must match on an input that can be read only once",
+            new TextEquals(input),
+            new Matches<>(new TextOf(new StringReader(input)))
+        ).affirm();
+    }
+
+    private static final class TextEquals extends TextMatcherEnvelope {
```

Comment:
@victornoel IMHO *all* classes should be ```public```. Why? Because if they're ```private``` you're obstructing the reuse of this functionalities by others parts. @paulodamaso WDYT?

Reply:
@fabriciofx it's a class that is part of the test: it's not meant to be reused. If someone wants that functionality, there is `TextIs`, but here the point is to te be self-sufficient with the envelope to test its behaviour. If there was any reason to reuse this class, then it would be as part of this project, since it's in the test sources, and in that case, when it is needed, we can always bring it outside of this class and then test it before making it public. But I wouldn't recommend it because it would defeat the purpose of having a self-sufficient test.

The real problem here is that `TextMatcherEnvelope` is an abstract class with behaviour that needs to be tested, this is not great, see #135.

Reply:
@fabriciofx @victornoel For me it's okay the way it is now; if we need to use this in he future, we can extract it.

Reply:
@fabriciofx are we good to merge then?

Reply:
@victornoel ok. @paulodamaso can you merge it now, please?